frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/breakdown-card/utils.tsx

```diff
@@ -95,24 +95,28 @@ export const sortInstantVectorStats = (stats: DataPoint[]): DataPoint[] => {
   return stats.length === 6 ? stats.splice(0, 5) : stats;
 };
 
-export const getStackChartStats: GetStackStats = (response, humanize) =>
+export const getStackChartStats: GetStackStats = (response, humanize, labelNames) =>
   response.map((r, i) => {
     const capacity = humanize(r.y).string;
     return {
       // x value needs to be same for single bar stack chart
       x: '0',
       y: r.y,
-      name: _.truncate(`${r.x}`, { length: 12 }),
-      link: `${r.x}`,
-      color: Colors.LINK,
+      name: labelNames ? labelNames[i] : _.truncate(`${r.x}`, { length: 12 }),
```

Comment:
A question here - Shouldn't you be checking `labelNames[i]` too here, before applying it in case `labelNames[i]` is undefined?

Reply:
No need as there a few cases where we pass label names and for those cases, we can put the correct number of items, in the array. I have created a map for those special cases.